lib/bazel/commands/commands.go

```diff
@@ -87,6 +91,84 @@ func NewAffectedTargetsList(parent *AffectedTargets) *AffectedTargetsList {
 }
 
 func (c *AffectedTargetsList) Run(cmd *cobra.Command, args []string) error {
+	// TODO(scott): Determine how the workspace root is found
+	gitRoot, err := findGitRoot()
+	if err != nil {
+		return fmt.Errorf("can't find git repo root: %v", err)
+	}
+
+	// Create temporary worktrees in which to execute bazel commands.
+	// If the end commit is not provided, use the current git directory as the end
+	// worktree, which will include uncommitted local changes.
+	startTree, err := git.NewTempWorktree(gitRoot, c.parent.Start)
+	if err != nil {
+		return fmt.Errorf("can't generate worktree for committish %q: %v", c.parent.Start, err)
+	}
+	defer startTree.Close()
+
+	endTreePath := gitRoot
+	if c.parent.End != "" {
+		endTree, err := git.NewTempWorktree(gitRoot, c.parent.End)
+		if err != nil {
+			return fmt.Errorf("can't generate worktree for committish %q: %v", c.parent.End, err)
+		}
+		defer endTree.Close()
+		endTreePath = endTree.Root()
+	}
+
+	// Open the bazel workspaces, using a well-known output_base. Since the
+	// temporary worktrees created above will have a different path on every
+	// invocation, by default bazel will create a new cache directory for them,
+	// re-download all dependencies, etc. which is both slow and will eventually
+	// fill up the disk. Reusing an output base location will ensure that at most
+	// only two are created for the purposes of this subcommand.
+	startOutputBase, err := cacheDir("affected_targets/start")
+	if err != nil {
+		return fmt.Errorf("failed to create output_base: %v", err)
+	}
+	endOutputBase, err := cacheDir("affected_targets/end")
+	if err != nil {
+		return fmt.Errorf("failed to create output_base: %v", err)
+	}
+	startWorkspace, err := bazel.OpenWorkspace(startTree.Root(), bazel.WithOutputBase(startOutputBase))
+	if err != nil {
+		return fmt.Errorf("failed to open bazel workspace for committish %q: %v", c.parent.Start, err)
+	}
+	endWorkspace, err := bazel.OpenWorkspace(endTreePath, bazel.WithOutputBase(endOutputBase))
+	if err != nil {
+		return fmt.Errorf("failed to open bazel workspace: %v", err)
+	}
+
+	// Get all target info for both VCS time points.
+	targets, err := startWorkspace.Query("deps(//...)", bazel.WithKeepGoing(), bazel.WithUnorderedOutput())
+	if err != nil {
+		return fmt.Errorf("failed to query deps for start point: %v", err)
+	}
+	// TODO(scott): Replace with logging
```

Comment:
yeah, I'd pass in a logger object to the function. Also consider that ... do we send the log to stderr? syslog? debug as well? a file on disk? ...

You can use: https://github.com/enfabrica/enkit/blob/master/lib/logger/klog/factory.go#L48 and friends to create / populate the object.

(in the main/app - in the main/app, though, it's fine to initialize and pass a glog object into the command initializer)